gtk/gtkwindow.c

```diff
@@ -673,6 +673,45 @@ startup_id_is_fake (const gchar* startup_id)
   return strncmp (startup_id, "_TIME", 5) == 0;
 }
 
+#ifdef GDK_WINDOWING_X11
+gboolean
+gtk_window_property_notify (GtkWidget *widget, GdkEventProperty *event)
```

Comment:
Oh, wait, I didn't notice this.

GtkWindow is *definitely* the wrong place to do this. No platform-specific code should go into GTK+.

You should add this code inside the X11 backend of GdkWindow.